tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -563,6 +595,44 @@ def rebuild_index(self):
             with self.context.context.track_errors() as _:
                 self._search_instantiations(asts)
 
+    def called_function_at_position(
+        self, position: Position
+    ) -> Optional[chapel.TypedSignature]:
+        """
+        Given a particular position, finds function being called at that
+        position.
+
+        Note: this function implies using resolution, and should only
+        be called if the resolver is enabled.
+        """
+
+        # Performance:
+        # since we don't have "call segments" (or segments for any other type
+        # of node), we have to iterate over all calls and check if they're in
+        # range. We can do better if we track segments for these.
```

Comment:
we definitely should track call segments, although no need to do that in this PR. Can you mark this as "Todo: Performance:" so that if grepping for todos this jumps out more?